devito/symbolics/printer.py

```diff
@@ -102,6 +102,10 @@ def _print_Mod(self, expr):
         args = ['(%s)' % self._print(a) for a in expr.args]
         return '%'.join(args)
 
+    def _print_Mul(self, expr):
+        term = super()._print_Mul(expr)
+        return term.replace("(-1)*", "-")
```

Comment:
I don't liks this -- the symptom points to a sympy-reconstruction bug somewhere

For example, I have a factorizer patch in my branch that fixes an identical symptom:

https://github.com/devitocodes/devito/commit/9a4a31a65143d29b572218a385a63b2c4c871bee#diff-a3ab417a5b298b20b70b45a84b15b8324f0526f6fe12a588a3f2912ed057d377R145

Does it help it here?

We surely shouldn't hack away the underlying sympy representation here, but rather fix the actual bug -- hopefully my patch+test does it!

Reply:
There is a small issue with some sympy version that pickling does not lead to the same expression between `-expr` and `(-1) * expr` not sure exactly why.

Reply:
What's the MFE? I can look into it

Reply:
`test_pickle.py` at `TestOperator::test_usave_sampled`

Reply:
I think you can probably cherry-pick this commit 

https://github.com/devitocodes/devito/pull/2183/commits/9a4a31a65143d29b572218a385a63b2c4c871bee

to get rid of this hack